### ISSUE TITLE
Geolocation only works if I use React Native Maps with showsUserLocation prop

### ISSUE THREAD

**Issue by Lea23VC** [Agontuk/react-native-geolocation-service#310]
The geolocation give bad coords and always give the same ones, they don't update, but if I add the "showsUserLocation" prop to the MapView from React Native Maps, the geolocation service starts to work without any problem.

I don't know what I'm doing wrong, the app has the permissions and I updated the android xml with <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />, <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />, <uses-permission android:name="android.permission.FOREGROUND_SERVICE" /> and even <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />

**Comment by hgvo9** [comment]
Any news @Lea23VC ? I'm having the same issues as you.

**Comment by p-ugulino** [comment]
Same problem here

**Comment by daniele-mc** [comment]
> The geolocation give bad coords and always give the same ones, they don't update, but if I add the "showsUserLocation" prop to the MapView from React Native Maps, the geolocation service starts to work without any problem.
> 
> I don't know what I'm doing wrong, the app has the permissions and I updated the android xml with , , and even

GOD

**Comment by hgvo9** [comment]
> Same problem here
> 
> update: managed to solve it by now with @Lea23VC solution

Thank you for your answer, I'm afraid I can't see @Lea23VC solution, could you please link it me ?

**Comment by p-ugulino** [comment]
> > Same problem here
> > update: managed to solve it by now with @Lea23VC solution
> 
> Thank you for your answer, I'm afraid I can't see @Lea23VC solution, could you please link it me ?

Hello! We managed to get it solved adding "showsUserLocation" on the MapView we have.

**Comment by Lea23VC** [comment]
Hello guys

My problem is that I can't use geolocation without "showsUserLocation", but want to use it without that prop. I want to use a custom marker, not that blue dot

**Comment by hgvo9** [comment]
> Hello guys
> 
> My problem is that I can't use geolocation without "showsUserLocation", but want to use it without that prop. I want to use a custom marker, not that blue dot

Yes indeed, I want to use a custom marker too, weird enough it works on iOS without the prop "showsUserLocation" but on Android it doesn't ...

**Comment by sdandois** [comment]
I think you are not asking for the ACCESS_FINE_LOCATION permission.

**Comment by CDBridger** [comment]
I have the same issue, have both permissions in the manifest but it doesn't actual provide the permissions.

```   
<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
 ```
 
 EDIT:
 
 Seems like you need to prompt the user with
 
 ```
 import { PermissionsAndroid } from 'react-native';
 

await PermissionsAndroid.requestMultiple([
   'android.permission.ACCESS_COARSE_LOCATION',
   'android.permission.ACCESS_FINE_LOCATION'
 ]);
```

should probably be added to the setup instructions, or part of the request authorization api call and NOT labelled iOS only.
